pkg/didcomm/protocol/issuecredential/service.go

```diff
@@ -168,6 +178,11 @@ func New(p Provider) (*Service, error) {
 	return svc, nil
 }
 
+// Use allows providing middlewares
+func (s *Service) Use(middlewares ...Middleware) {
+	s.middlewares = append(s.middlewares, middlewares...)
```

Comment:
suggest you use [`stack`](https://pkg.go.dev/github.com/golang-collections/collections/stack) here so that the ordering is clear and that for-loop where you build the handler chain makes more sense

Reply:
Changed to:
```
func (s *Service) Use(items ...Middleware) {
	s.stack = append(s.stack, items...)
}
```

Reply:
I meant to use a `stack` as a data structure (hence the link to https://pkg.go.dev/github.com/golang-collections/collections/stack)....

Reply:
Oh, I got it :) I don't think we really need that. This is one more dependency for the project.
But I did a small refactor. And now it undesirable.
```
func (s *Service) Use(items ...Middleware) {
	var handler Handler = initialHandler
	for i := len(items) - 1; i >= 0; i-- {
		handler = items[i](handler)
	}

	s.middleware = handler
}
```